webserver/webserver/src/main/java/io/helidon/webserver/ForwardingHandler.java

```diff
@@ -310,6 +311,10 @@ private boolean channelReadHttpRequest(ChannelHandlerContext ctx, Context reques
         request.headers().remove(Http.Header.X_HELIDON_CN);
         Optional.ofNullable(ctx.channel().attr(CLIENT_CERTIFICATE_NAME).get())
                 .ifPresent(name -> request.headers().set(Http.Header.X_HELIDON_CN, name));
+        // If the client x509 certificate is present on the channel, add it to the context scope of the ongoing
+        // request so that helidon handlers can inspect and react to this.
+        Optional.ofNullable(ctx.channel().attr(CLIENT_CERTIFICATE).get())
```

Comment:
We have chosen an approach with header (as can be seen right above with `X_HELIDON_CN`). I think that to be consistent, the certificate should be added as a header as well. @Verdent please comment as well

Reply:
currently it's a java.security X509Certificate

Reply:
The problem here is that the cert is currently java.security.X509Certificate. If we would use headers to store it, that would mean, we need to create String out of it first and when if anyone wants to use that, one would need to create X509Cert out of the String again.

There is header method which supports Object value, but when I went through the code, I saw it is actually converted to the String later on in Netty. (It is done by simply calling toString method). Because of that, I might say it is more convenient to do it as @astromechza did that.

Reply:
Yeah serializing back and forth to string is going to be fairly expensive and painful compared to having the object there in memory.

Reply:
one note on the existing X_HELIDON_CN header is that it is not documented anywhere and only used in some test cases.

Reply:
OK. in that case we should add the CN in a similar way, we may create a follow up issue.